zerver/lib/url_preview/parsers/generic.py

```diff
@@ -47,7 +47,12 @@ def _get_image(self) -> Optional[str]:
         soup = self._soup
         first_h1 = soup.find("h1")
         if first_h1:
+            # First look for image after
             first_image = first_h1.find_next_sibling("img", src=True)
+            # If there is no image after, look for image before
```

Comment:
This comment explains what the code is doing; but that's pretty clear from skimming the code. What we like to see in comments is a prose explanation of __why__ we're doing this. Probably just rewriting the block quote at the start of the function to talk about how we're heuristically trying to guess what the main image is, and we look for images both before and after the first heading, with some reasoning for why we should prefer the first image after.